scripts/boxgame.js

```diff
@@ -0,0 +1,119 @@
+class BoxGameComponent {
+    box = {};
+    score = document.createElement("div");
+    squares = [];
+    container = document.createElement('div');
+    gameAreaEl = document.createElement("div");
+    gamebox = {
+        x: Math.floor(800 / 100),
+        y: Math.floor(200 / 100)
+    };
+    player = {
+        speed: 100,
+        square: 1,
+        score: 0
+    };
+
+    init = (button) => {
+        this.gameAreaEl.innerHTML = '';
+        this.container.innerHTML = '';
+        this.score.innerHTML = '';
+        document.body.appendChild(this.container);
+        this.createGameAreaDom();
+        this.addListeners();
+        this.build();
+        button.style.display = 'none';
+    };
+
+    addListeners = () => {
+        document.addEventListener('DOMContentLoaded', this.build);
+        document.addEventListener('keyup', function (e) {
+            const allowKey = {
+                37:'left',
+                38: 'up',
+                39: 'right',
+                40: 'down'
+            };
+            if(allowKey[e.keyCode]) {
+                game.handleKey(allowKey[e.keyCode])
+            }
+        });
+    };
+
+    createGameAreaDom = () => {
+        this.gameAreaEl.setAttribute("class", "gameArea");
+        this.score.setAttribute("class", "score");
+        this.container.appendChild(this.gameAreaEl);
+        this.container.appendChild(this.score);
+        this.gameAreaEl.getBoundingClientRect();
+    };
+
+    setGameArea() {
+        return this.gameAreaEl.getBoundingClientRect();
+    };
+
+    makeActive = () => {
+        let randomIndex = Math.floor(Math.random() * this.squares.length);
+
+        if (randomIndex != 0 && this.player.square != randomIndex){
+            this.squares[randomIndex].classList.add('active');
+        } else {
+            this.makeActive();
+        }
+
+    };
+
+    handleKey = (key) => {
```

Comment:
i think this function needs refactoring and the position of the user square should be tracked in an abstract way 
you can then assign css values to create the green and red css states of the visual boxes based on the outcome of this logic. This helps separate the game logic from the presentation of the DOM elements
this would make it easier to test which is also why i suspect there are no unit tests for this function.